volume-group-operator/controllers/volumegroup_controller.go

```diff
@@ -18,24 +18,26 @@ package controllers
 
```

Comment:
where do we handle the events we should add to the `volumeGroup` objects in Kubernetes, both in success and failure?

Reply:
in the `Error` field in `Status`

Reply:
this is not event, if there is a newer Error the user won't see the previous ones

Reply:
there is a difference between field in the Status than Event of an object

Reply:
the user won't see any event in the long run, they will all disappear. the Error in Status will remain till resolved.
anyway, If you find a way to add Events easily fill free to add them, I didn't.

Reply:
it's part of the spec and part of your ticket, you don't create objects and don't create events for them, it's a blocker for this PR IMO

Reply:
[this](https://pkg.go.dev/k8s.io/client-go/tools/events#EventSinkImpl.Create) doesn't work?

Reply:
I managed to create events [here](https://github.com/IBM/ibm-block-csi-operator/blob/feature/CSI-4972_pvc_controller_remove_pvc_from_vg_with_grpc/volume-group-operator/controllers/utils/event.go) you can use this code

Reply:
please link the spec requirement for the events.

Reply:
I opened a ticket for it CSI-5149

Reply:
ticket is not good, we need to do it in this version, and it's known thing for every Kubernetes object, you can't create one without events

Reply:
if the ticket will be in 1.11 version so it's fine, but if not, we can't GA that way

Reply:
it is, but anyway I disagree that it's blocking. its part of no spec and replication sidecar doesn't even have more events than the Error in status

Reply:
if the replication sidecar doesn't do it it doesn't mean it is good, we need to be better than them, and I don't think we can GA this feature without events so we must do this ticket and it's not that big of a change because I did most of the job already

Reply:
Arbel opened ticket for adding events and this ticket is important enough to handle it until GA. We still have 2 sprints to do it. I don't think this discussion is productive, let's merge this PR and then Arbel will be able to proceed with events ticket.

Reply:
feature freeze is in this sprint, so no new content after this sprint

Reply:
so we don't have two sprints for it

Reply:
feature freeze is in this sprint can't happen, unless we decided not to come with replication for volume groups

Reply:
currently, the feature freeze is in this sprint and we should act like this until Eran will change it